src/pages/scheduler/components/ShareTimetableButton.tsx

```diff
@@ -0,0 +1,185 @@
+import { getReadableTerm } from 'lib/utils';
+
+const InformationText = (props: { isSmallScreen: boolean; term: string }) => {
+  return (
+    <Alert status="info" borderRadius="10px">
+      {props.isSmallScreen ? <AlertIcon /> : null}
+      We've generated a link that you can share to allow people to view, compare, and import the courses and sections
+      you currently have selected for the {getReadableTerm(props.term)} term.
+    </Alert>
+  );
+};
+
+const ShareCourseCard = (props: { course: SavedCourse }) => {
+  return (
+    <Flex height="100%" direction="column">
+      <HStack w="100%" bg="#EDF2F7" justifyContent="center" p="0.2em" borderTopRadius="4px" padding="4px">
+        <Heading size="xs">
+          {props.course.subject} {props.course.code}
+        </Heading>
+      </HStack>
+      <VStack bg={props.course.color} flex={1} justifyContent="center" borderBottomRadius="4px" padding="4px">
+        <Heading justifyContent="center" size="sm">
+          {props.course.lecture} {props.course.lab} {props.course.tutorial}
+        </Heading>
+      </VStack>
+    </Flex>
+  );
+};
+
+const SelectedCoursesTable = (props: { courses: Array<SavedCourse>; term: string }) => {
+  return (
+    <Wrap variant="striped">
+      {props.courses.length > 0 ? (
+        props.courses.map((course) => {
+          if (course.lecture || course.lab || course.tutorial) {
+            return (
+              <WrapItem>
+                <ShareCourseCard course={course} />
+              </WrapItem>
+            );
+          }
+          return null;
+        })
+      ) : (
+        <Text>
+          Unable to find saved courses for <Box as="strong"> {getReadableTerm(props.term || '')} </Box>
+        </Text>
+      )}
+    </Wrap>
+  );
+};
+
+const SocialMediaButtons = (props: { slug: string }) => {
+  return (
+    <HStack justify="center">
+      <EmailShareButton children={<EmailIcon size={50} round={true} />} url={props.slug} />
+      <FacebookShareButton children={<FacebookIcon size={50} round={true} />} url={props.slug} />
+      <TelegramShareButton children={<TelegramIcon size={50} round={true} />} url={props.slug} />
+      <WhatsappShareButton children={<WhatsappIcon size={50} round={true} />} url={props.slug} />
+      <TwitterShareButton children={<TwitterIcon size={50} round={true} />} url={props.slug} />
+    </HStack>
+  );
+};
+
+const CopyLinkUrl = (props: { isSmallScreen: boolean; slug: string }) => {
+  const toast = useToast();
+  return (
+    <HStack justify="space-between" padding="5px" borderWidth="1px" borderColor="gray.300">
+      <HStack justify="center" flexGrow={4}>
+        {props.isSmallScreen ? <Icon boxSize="1.25em" as={HiLink} /> : undefined}
+        <Input id="timetable_slug" value={props.slug} variant="filled" isReadOnly={true} />
+      </HStack>
+      <Button
+        size={props.isSmallScreen ? 'sm' : 'md'}
+        colorScheme="blue"
+        leftIcon={<CopyIcon />}
+        onClick={() => {
```

Comment:
Typically we don't want to use arrow functions here and would probably define a function above called `handleClick` but since there is still API integration to go into slug generation, we can refactor this later